### ISSUE TITLE
JDK20  jdk_nio_0_FAILED UnsupportedOperationException: Unimplemented on AIX

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#16970]
Failure link
------------

From [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk20_j9_extended.openjdk_ppc64_aix_Personal/1/)(`paix906`):
```
09:32:18  openjdk version "20-internal" 2023-03-21
09:32:18  OpenJDK Runtime Environment (build 20-internal-adhoc.jenkins.BuildJDK20ppc64aixPersonal)
09:32:18  Eclipse OpenJ9 VM (build master-8cc902bc75f, JRE 20 AIX ppc64-64-Bit Compressed References 20230316_18 (JIT enabled, AOT enabled)
09:32:18  OpenJ9   - 8cc902bc75f
09:32:18  OMR      - 035ec68ec0c
09:32:18  JCL      - 10a35739269 based on jdk-20+36)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&TARGET=extended.openjdk&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=&VENDOR_TEST_DIRS=functional&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=adoptium%3Amaster&OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&PLATFORM=ppc64_aix&GENERATE_JOBS=true&KEEP_REPORTDIR=true&PERSONAL_BUILD=false&DOCKER_REGISTRY_DIR=&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&RERUN_ITERATIONS=3&SETUP_JCK_RUN=false&DOCKER_REGISTRY_URL_CREDENTIAL_ID=&LABEL=&EXTRA_OPTIONS=&BUILD_IDENTIFIER=fengj%40ca.ibm.com&CUSTOMIZED_SDK_URL=https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2FBuild_JDK20_ppc64_aix_Personal%2F18%2FOpenJ9-JDK20-ppc64_aix-20230316-082142.tar.gz+https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2FBuild_JDK20_ppc64_aix_Personal%2F18%2Ftest-images.tar.gz&JENKINS_KEY=Jenkins+4096+key&ADOPTOPENJDK_BRANCH=master&LIGHT_WEIGHT_CHECKOUT=true&USE_JRE=false&ARTIFACTORY_SERVER=na-public.artifactory.swg-devops&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&JDK_VERSION=20&DOCKER_REGISTRY_URL=&ITERATIONS=1&VENDOR_TEST_REPOS=git%40github.ibm.com%3Aruntimes%2Ftest.git&JDK_REPO=git%40github.com%3Aibmruntimes%2Fopenj9-openjdk-jdk20.git&JCK_GIT_BRANCH=main&OPENJ9_BRANCH=master&OPENJ9_SHA=8cc902bc75ffa8e24f4beef2a1f8726a383771eb&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=master&OPENJ9_REPO=git%40github.com%3Aeclipse%2Fopenj9.git&UPSTREAM_JOB_NAME=&CLOUD_PROVIDER=&CUSTOM_TARGET=&VENDOR_TEST_SHAS=034d4edd4350276a2d01aade526b242fb70f4eee&JDK_BRANCH=openj9&LABEL_ADDITION=ci.project.openj9&ARTIFACTORY_REPO=sys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com&ARTIFACTORY_ROOT_DIR=&UPSTREAM_TEST_JOB_NUMBER=&DOCKERIMAGE_TAG=&JDK_IMPL=openj9&TEST_TIME=&SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&AUTO_DETECT=true&SLACK_CHANNEL=%23rt-jenkins&DYNAMIC_COMPILE=false&RELATED_NODES=&ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&APPLICATION_OPTIONS=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=7c1c2c28-650f-49e0-afd1-ca6b60479546&ARCHIVE_TEST_RESULTS=false&NUM_MACHINES=&OPENJDK_SHA=&TRSS_URL=http%3A%2F%2Ftrss1.fyre.ibm.com&USE_TESTENV_PROPERTIES=false&BUILD_LIST=openjdk&UPSTREAM_JOB_NUMBER=&STF_OWNER_BRANCH=adoptium%3Amaster&TIME_LIMIT=20&JVM_OPTIONS=&PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
10:29:17  variation: Mode150
10:29:17  JVM_OPTIONS:  -XX:+UseCompressedOops 

10:54:45  TEST: java/nio/channels/vthread/BlockingChannelOps.java#default

10:54:45  STDERR:
10:54:45  java.lang.UnsupportedOperationException: Unimplemented on AIX
10:54:45  	at java.base/sun.nio.ch.PollsetPoller.poll(PollsetPoller.java:60)
10:54:45  	at java.base/sun.nio.ch.Poller.poll(Poller.java:363)
10:54:45  	at java.base/sun.nio.ch.Poller.pollLoop(Poller.java:270)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  	at java.base/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)
10:54:45  java.lang.UnsupportedOperationException: Unimplemented on AIX
10:54:45  	at java.base/sun.nio.ch.PollsetPoller.poll(PollsetPoller.java:60)
10:54:45  	at java.base/sun.nio.ch.Poller.poll(Poller.java:363)
10:54:45  	at java.base/sun.nio.ch.Poller.pollLoop(Poller.java:270)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  	at java.base/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)
10:54:45  java.lang.UnsupportedOperationException: Unimplemented on AIX
10:54:45  	at java.base/sun.nio.ch.PollsetPoller.implRegister(PollsetPoller.java:48)
10:54:45  	at java.base/sun.nio.ch.Poller.updateLoop(Poller.java:289)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  	at java.base/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)

10:54:45  TEST: java/nio/channels/vthread/BlockingChannelOps.java#direct-register

10:54:45  STDERR:
10:54:45  java.lang.UnsupportedOperationException: Unimplemented on AIXjava.lang.UnsupportedOperationException: Unimplemented on AIX
10:54:45  
10:54:45  	at java.base/sun.nio.ch.PollsetPoller.poll(PollsetPoller.java:60)
10:54:45  	at java.base/sun.nio.ch.Poller.poll(Poller.java:363)
10:54:45  	at java.base/sun.nio.ch.PollsetPoller.poll(PollsetPoller.java:60)
10:54:45  	at java.base/sun.nio.ch.Poller.poll(Poller.java:363)
10:54:45  	at java.base/sun.nio.ch.Poller.pollLoop(Poller.java:270)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  	at java.base/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)
10:54:45  	at java.base/sun.nio.ch.Poller.pollLoop(Poller.java:270)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  	at java.base/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:186)
10:54:45  java.lang.Exception: failures: 28
10:54:45  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:95)
10:54:45  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:53)
10:54:45  	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
10:54:45  	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
10:54:45  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:125)
10:54:45  	at java.base/java.lang.Thread.run(Thread.java:1639)
10:54:45  
10:54:45  JavaTest Message: Test threw exception: java.lang.Exception: failures: 28

10:54:45  --------------------------------------------------
11:10:45  Test results: passed: 388; failed: 1; error: 1
11:10:51  Report written to /home/jenkins/workspace/Test_openjdk20_j9_extended.openjdk_ppc64_aix_Personal/aqa-tests/TKG/output_16789737008877/jdk_nio_0/report/html/report.html
11:10:51  Results written to /home/jenkins/workspace/Test_openjdk20_j9_extended.openjdk_ppc64_aix_Personal/aqa-tests/TKG/output_16789737008877/jdk_nio_0/work
11:10:51  Error: Some tests failed or other problems occurred.
11:10:51  -----------------------------------
11:10:51  jdk_nio_0_FAILED
```

**Comment by pshipton** [comment]
@backwaterred pls take a look.

**Comment by backwaterred** [comment]
I know this one already; I actually wrote that failure text when the Loom changes first merged.

I have an implementation complete for this portion of it, but I will need a bit more time to complete the rest of the VThread changes before merging.

**Comment by tajila** [comment]
@backwaterred Any update on this?

**Comment by backwaterred** [comment]
Hi @tajila, a quick update. I have a PR open for this issue upstream. I am waiting for one other community review, and then I am good to go. I will backport to 20 once it merges.

**Comment by tajila** [comment]
thanks @backwaterred

**Comment by pshipton** [comment]
The backport for 20 is https://github.com/openjdk/jdk20u/pull/84

For testing, ported to OpenJ9 in https://github.com/pshipton/openj9-openjdk-jdk20/commit/fae5801febe15f14fa0dd85272c8362e76a775a9

java/nio/channels/vthread/BlockingChannelOps.java passes on OpenJ9 with this update.

For now, since we are waiting for a PingPong test fix, we can wait for the [OpenJDK PR](https://github.com/openjdk/jdk20u/pull/84) to be merged, or to see if there are any review comments. If this becomes the only remaining issue, we'll take the fix as-is to the 0.39 release branch.
